internal/scan/scan.go

```diff
@@ -70,27 +73,129 @@ type EventApprovalForAll struct {
 	Approved bool
 }
 
+// EventNewERC721BridgelessMinting is the ERC721 event emitted when a new Bridgeless Minting contract is deployed
+type EventNewERC721BridgelessMinting struct {
+	NewContractAddress common.Address
+	BaseURI            string
+}
+
 // Scanner is responsible for scanning and retrieving the ERC721 events
 type Scanner interface {
+	ScanNewBridgelessMintingEvents(ctx context.Context, fromBlock, toBlock *big.Int) error
 	ScanEvents(ctx context.Context, fromBlock *big.Int, toBlock *big.Int) ([]Event, error)
 }
 
 type scanner struct {
-	client   EthClient
-	contract common.Address
+	client    EthClient
+	contracts []common.Address
+	storage   Storage
 }
 
 // NewScanner instantiates the default implementation for the Scanner interface
-func NewScanner(client EthClient, contract common.Address) Scanner {
-	return scanner{
-		client:   client,
-		contract: contract,
+func NewScanner(client EthClient, s Storage, contracts ...string) Scanner {
+	scan := scanner{
+		client:  client,
+		storage: s,
 	}
+	for _, c := range contracts {
+		scan.contracts = append(scan.contracts, common.HexToAddress(c))
+	}
+	return scan
+}
+
+// ScanEvents returns the ERC721 events between fromBlock and toBlock
+func (s scanner) ScanNewBridgelessMintingEvents(ctx context.Context, fromBlock, toBlock *big.Int) error {
```

Comment:
maybe this function should return array of events. And another function that deals with saving events should save them (either in db or file or not save them at all).